registry/storage/shares.go

```diff
@@ -395,7 +398,20 @@ func (s *sharesStorage) UpdateValidatorsMetadata(data map[spectypes.ValidatorPK]
 		return err
 	}
 
-	return s.saveToDB(nil, shares...)
+	if len(shares) > 0 {
+		return s.saveToDB(nil, shares...)
+	}
+
+	return nil
+}
+
+func extractMetadata(share *types.SSVShare) *beaconprotocol.ValidatorMetadata {
```

Comment:
what do you think about making it a method on `SSVShare`?

Reply:
done here https://github.com/ssvlabs/ssv/pull/2155#issuecomment-2817073233